modules/plugins/installer.py

```diff
@@ -0,0 +1,130 @@
+import ez
+
+from pathlib import Path
+from zipfile import ZipFile
+from yaml import load, Loader
+from shutil import rmtree
+
+from pydantic import BaseModel
+
+from utilities.semver import SemanticVersion
+
+from .manager import PluginManager
+
+
+def read_manifest(path: Path | str):
+    if isinstance(path, str):
+        path = Path(path)
+    manifest_file = path / PluginInstaller.MANIFEST_FILENAME
+
+    if not manifest_file.exists():
+        raise FileNotFoundError(f"Manifest file not found in {path}")
+
+    with manifest_file.open("r") as file:
+        data = load(file, Loader=Loader)
+    for key in data:
+        if '-' in key:
+            new_key = key.replace('-', '_')
+            data[new_key] = data.pop(key)
+    manifest = PluginManifest.model_validate(data)
+
+    return manifest
+
+
+def move_file(src: Path, dest: Path, name: str = None):
+    dest.mkdir(parents=True, exist_ok=True)
+    if name is None:
+        name = src.name
+    src.rename(dest / name)
+
+    print(f"Moved {src} to {dest / name}")
```

Comment:
Don't use `print` in this project, if you want to log something use `ez.log`

Reply:
I didn't

Reply:
Lies